statistics/boostrap.go

```diff
@@ -54,6 +54,7 @@ func (h *Handle) initStatsMeta(is infoschema.InfoSchema) (statsCache, error) {
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
+	defer terror.Call(rc[0].Close)
```

Comment:
We should move the defer to line 54.

Reply:
But if the err is not nil, `rc` will be nil.

Reply:
We can add a check as following:
```
if len(rc) > 0{ // defer ... }
```

Reply:
How about if rc != nil {rc[0].Close()}